#! /bin/sh /usr/share/dpatch/dpatch-run
## 02cachedesync.dpatch by  <az@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix for (upstream)#497243: remove causes cache desynchronization

@DPATCH@
diff -urNad duplicity-0.6.06~/src/collections.py duplicity-0.6.06/src/collections.py
--- duplicity-0.6.06~/src/collections.py	2009-10-30 06:51:44.000000000 +1000
+++ duplicity-0.6.06/src/collections.py	2010-01-26 16:58:00.267288762 +1000
@@ -145,7 +145,12 @@
             if (pr
                 and pr.time == self.time
                 and pr.start_time == self.start_time
-                and pr.end_time == self.end_time):
+                and pr.end_time == self.end_time
+                and pr.type != "new-sig" ):
+                # do not remove new sigs from the cache:
+                # they aren't removed from the remote archive,
+                # and subsequent backups will have to resync
+                # which is bad if running non-interactive with encrypt-key 
                 try:
                     globals.archive_dir.append(lfn).delete()
                 except:
